metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -878,8 +877,17 @@ class MetalsLspService(
 
   def connectTables(): Connection = tables.connect()
 
-  def initialized(): Future[Unit] =
+  def loadProjectRootSetting(): Future[Unit] = Future {
+    for {
+      relativePath <- tables.projectRoot.relativePath()
+      absolutePath = folder.resolve(relativePath)
+      if (absolutePath.isDirectory)
+    } buildTools.setProjectRoot(absolutePath)
```

Comment:
I wonder if we should save the state. Potentially, one could change the location of build.sbt file and in that case the project root would become invalid. Ideally, we would calculate it every time but with a limited depth?

Reply:
We do calculate it every time. It's only saved and treated as source of truth if it's set by the user explicitly.

Reply:
ach alright! I didn't notice